From 70ec1af9b3c5682428fb7a955d952be46ad679d5 Mon Sep 17 00:00:00 2001
From: akodanka <anoob.anto.kodankandath@intel.com>
Date: Sat, 24 Jul 2021 18:44:16 +0530
Subject: [PATCH] Disable flag useFlexibleYuv, as it adds delay for GPU buffers

Even though support for HAL_PIXEL_FORMAT_YCrCb_420_SP and
HAL_PIXEL_FORMAT_YV12 are added to HAL, the fag "useFlexibleYuv"
would request for HAL_PIXEL_FORMAT_YCbCr_420_888 format, and do
the conversion at CallbackProcessor::convertFromFlexibleYuv.
When doing this conversion on GPU buffers, it adds latency to
the buffer processing. Hence, disabling this flag at Framework.

Change-Id: I9ef665ffdf63b7c8ff63f889aa8702505a78fd12
Tracked-On: OAM-99757
Signed-off-by: akodanka <anoob.anto.kodankandath@intel.com>
---
 .../camera/libcameraservice/api1/client2/Parameters.cpp     | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/api1/client2/Parameters.cpp b/services/camera/libcameraservice/api1/client2/Parameters.cpp
index 28d186abb4..2334dc4bc1 100644
--- a/services/camera/libcameraservice/api1/client2/Parameters.cpp
+++ b/services/camera/libcameraservice/api1/client2/Parameters.cpp
@@ -1194,12 +1194,16 @@ status_t Parameters::buildFastInfo(CameraDeviceBase *device) {
 
     // Check if the HAL supports HAL_PIXEL_FORMAT_YCbCr_420_888
     fastInfo.useFlexibleYuv = false;
+    /* Disabled FlexibleYuv support, as it adds delay when using GPU buffers.
+     * Even though support for 420_SP and YV12 formats are added to HAL,
+     * this flag "useFlexibleYuv" would change the request format to 420_888,
+     * for these formats.
     for (size_t i = 0; i < availableFormats.size(); i++) {
         if (availableFormats[i] == HAL_PIXEL_FORMAT_YCbCr_420_888) {
             fastInfo.useFlexibleYuv = true;
             break;
         }
-    }
+    }*/
     ALOGV("Camera %d: Flexible YUV %s supported",
             cameraId, fastInfo.useFlexibleYuv ? "is" : "is not");
 
-- 
2.33.1

